Haskell/Blarney/Core/Net.hs

```diff
@@ -245,6 +247,80 @@ inlineSingleRefNet nl = do
+  (net { netPrim = Const 1 1, netInputs = [] }, True)
+zeroWidthNetTransform net@Net{ netPrim = NotEqual 0 } =
+  (net { netPrim = Const 1 0, netInputs = [] }, True)
+zeroWidthNetTransform net@Net{ netPrim = LessThan 0 } =
+  (net { netPrim = Const 1 0, netInputs = [] }, True)
+zeroWidthNetTransform net@Net{ netPrim = LessThanEq 0 } =
+  (net { netPrim = Const 1 1, netInputs = [] }, True)
+zeroWidthNetTransform net@Net{ netPrim = ZeroExtend 0 w } =
+  (net { netPrim = Const w 0, netInputs = [] }, True)
+zeroWidthNetTransform net@Net{ netPrim = SignExtend 0 w } =
+  (net { netPrim = DontCare w, netInputs = [] }, True)
+zeroWidthNetTransform net@Net{ netPrim = SelectBits 0 hi lo } =
+  (net { netPrim = DontCare (hi-lo), netInputs = [] }, True)
+zeroWidthNetTransform net@Net{ netPrim = Concat w0 w1, netInputs = [i0, i1] }
+  | w0 == 0 && w1 /= 0 = (net { netPrim = Identity w1, netInputs = [i1] }, True)
+  | w0 /= 0 && w1 == 0 = (net { netPrim = Identity w0, netInputs = [i0] }, True)
+  | otherwise = (net, False)
+zeroWidthNetTransform net@Net{ netPrim = Display args, netInputs = inpts } =
+  let bitArgs = [ ba | ba@(DisplayArgBit w) <- args ]
+      f (DisplayArgBit 0) i = (InputTree (DontCare 1) [], True)
+      f _ i = (i, False)
+      (tmps, changes) = unzip $ zipWith f bitArgs (tail inpts)
+      inpts' = head inpts : tmps
+  in (net { netInputs = inpts' }, or changes)
+-- TODO currently unsupported cases that could be transformed
+--zeroWidthNetTransform net@Net{ netPrim = CountOnes _ } =
+--  error "zeroWidthNetTransform unsupported on CountOnes Prim"
+--zeroWidthNetTransform net@Net{ netPrim = BRAM {} } =
+--  error "zeroWidthNetTransform unsupported on BRAM Prim"
+--zeroWidthNetTransform net@Net{ netPrim = TrueDualBRAM {} } =
+--  error "zeroWidthNetTransform unsupported on TrueDualBRAM Prim"
+--zeroWidthNetTransform net@Net{ netPrim = Custom {} } =
+--  error "zeroWidthNetTransform unsupported on Custom Prim"
+--zeroWidthNetTransform net@Net{ netPrim = RegFileRead {} } =
+--  error "zeroWidthNetTransform unsupported on RegFileRead Prim"
+-- do nothing cases for all others
+zeroWidthNetTransform net = (net, False)
+
+-- | Tell if a 'Net' is a zero-width root 'Net' (in practice, only the 'Input'
+--   and 'Output' 'Prim's)
+isZeroWidthRootNet :: Net -> Bool
+isZeroWidthRootNet Net{ netIsRoot = True
+                      , netPrim = Input 0 _ } = True
+isZeroWidthRootNet Net{ netIsRoot = True
+                      , netPrim = Output 0 _ } = True
+isZeroWidthRootNet _ = False
+
+-- | Ignore 0-width Nets pass
+ignoreZeroWidthNet :: MNetlist -> IO Bool
+ignoreZeroWidthNet nl = do
+  pairs <- getAssocs nl -- list of nets with their index
+  changed <- newIORef False -- keep track of modifications to the 'Netlist'
+  -- For each 'Net' (in particular, those with a non-zero-width output) with at
+  -- least a zero-width input, transform it into a 'Net' with no reference to
+  -- any other zero-width output Net.
```

Comment:
For each 'Net' (in particular, those with a non-zero-width output) with at
least **one** zero-width input, transform it into a 'Net' with no reference to
any ~~other~~ zero-width output Net.